src/main/java/com/barszcz/server/controller/DevicesController.java

```diff
@@ -0,0 +1,151 @@
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class DevicesController {
+
+    private final ConfigurationDao configurationDao;
+    private final SimpMessagingTemplate simpMessagingTemplate;
+    private final DeviceTypeDao deviceTypeDao;
+
+    @GetMapping(path = "/find")
+    public List<ConfigurationModel> getAllDevices() {
+        return (List<ConfigurationModel>) configurationDao.findAll();
+    }
+
+    @GetMapping(path = "/findByIp")
+    public Optional<ConfigurationModel> getDeviceByIp(@RequestParam String ip) {
+        return configurationDao.findConfigurationModelByIpLike(ip);
+    }
+
+    @GetMapping(path = "/findByRoom")
+    public List<ConfigurationModel> getDevicesFromRoom(@RequestParam String room) {
+        return configurationDao.findConfigurationModelByRoomLike(room);
+    }
+
+    @PutMapping(path = "/changeLedState")
+    public void putLedState(@RequestParam String ip, int red, int green, int blue) throws Exception {
+        configurationDao.findConfigurationModelByIpLike(ip).map(configurationModel -> {
+            configurationModel.setRed(red);
+            configurationModel.setGreen(green);
+            configurationModel.setBlue(blue);
+            return configurationDao.save(configurationModel);
+        }).orElseThrow(
+                Exception::new
+        );
+    }
+
+    @PutMapping(path = "/changeState")
+    public void putState(@RequestParam String ip, String state) throws Exception {
+        configurationDao.findConfigurationModelByIpLike(ip).map(configurationModel -> {
+            configurationModel.setDeviceState(state);
+            return configurationDao.save(configurationModel);
+        }).orElseThrow(
+                Exception::new
+        );
+    }
+
+    @GetMapping(path = "/delete")
+    public List<ConfigurationModel> delete() {
+        configurationDao.deleteAll();
+        return (List<ConfigurationModel>) configurationDao.findAll();
+    }
+
+    @DeleteMapping(path = "/deleteDevice")
+    public void deleteByDeviceName(@RequestParam String name) {
+        configurationDao.deleteByDeviceNameLike(name);
+    }
+
+
+    @MessageMapping("/startup/{ip}")
+    public void startupDevice(@DestinationVariable("ip") String ip) {
+        if (configurationDao.findConfigurationModelByIpLike(ip).isPresent()) {
+            simpMessagingTemplate.convertAndSend("/device/" + ip, configurationDao.findConfigurationModelByIpLike(ip));
+        } else {
+            simpMessagingTemplate.convertAndSend("/device/newDevice", ip);
+        }
+    }
+
+    @MessageMapping("/addDevice/{ip}/{deviceName}/{room}/{deviceType}")
+    public void addDevice(@DestinationVariable("ip") String ip, @DestinationVariable("deviceName") String deviceName, @DestinationVariable("room") String room, @DestinationVariable("deviceType") String deviceType) {
```

Comment:
Tak to nie może wyglądać. stwórz klasę z tymi parametrami i załatw to adnotacją @RequestBody